snuba/query/processors/logical/filter_in_select_optimizer.py

```diff
@@ -1,234 +1,145 @@
-        expressions = map(lambda x: x.expression, query.get_selected_columns())
-        target_exp = None
-        for exp in expressions:
-            if self._contains_conditional_aggregate(exp):
-                if target_exp is not None:
+        # find and grab all the conditional aggregate functions
+        cond_agg_functions: list[FunctionCall | CurriedFunctionCall] = []
+        for selected_exp in query.get_selected_columns():
+            exp = selected_exp.expression
+            finder = FindConditionalAggregateFunctionsVisitor()
+            exp.accept(finder)
+            found = finder.get_matches()
+            if len(found) > 0:
+                if len(cond_agg_functions) > 0:
                     raise ValueError(
-                        "Was expecting only 1 select expression to contain condition aggregate but found multiple"
+                        "expected only one selected column to contain conditional aggregate functions but found multiple"
                     )
                 else:
-                    target_exp = exp
-
-        if target_exp is not None:
-            domains = self._get_conditional_domains(target_exp)
-            if len(domains) == 0:
-                raise ValueError("This shouldnt happen bc there is a target_exp")
-
-            # find the intersect of keys, across the domains of all conditional aggregates
-            key_intersect = set(domains[0].keys())
-            for i in range(1, len(domains)):
-                domain = domains[i]
-                key_intersect = key_intersect.intersection(set(domains[i].keys()))
-
-            # union the domains
-            domain_union: Domain = {}
-            for key in key_intersect:
-                domain_union[key] = set()
-                for domain in domains:
-                    domain_union[key] = domain_union[key].union(domain[key])
-
-            return domain_union
-        else:
-            return {}
+                    cond_agg_functions = found
 
-    def _contains_conditional_aggregate(self, exp: Expression) -> bool:
-        if isinstance(exp, FunctionCall):
-            if exp.function_name[-2:] == "If":
-                return True
-            for param in exp.parameters:
-                if self._contains_conditional_aggregate(param):
-                    return True
-            return False
-        elif isinstance(exp, CurriedFunctionCall):
-            if exp.internal_function.function_name[-2:] == "If":
-                return True
-            return False
-        else:
-            return False
+        if len(cond_agg_functions) == 0:
+            return None
 
-    def _get_conditional_domains(self, exp: Expression) -> list[Domain]:
-        domains: list[Domain] = []
-        self._get_conditional_domains_helper(exp, domains)
-        return domains
+        # validate the functions, and lift their conditions into new_condition, return it
+        new_condition = None
+        for func in cond_agg_functions:
```

Comment:
No, there can be multiple

`divide(sumIf(...), avgIf(...))` would cause it to be `[sumIf(...), avgIf(...)]`

I improved the comment for the visitor, hopefully this makes it a bit more clear.